compact_str/src/repr/mod.rs

```diff
@@ -346,6 +353,10 @@ impl Repr {
     /// Returns the string content, and only the string content, as a slice of bytes.
     #[inline]
     pub fn as_slice(&self) -> &[u8] {
+        if let Some(s) = self.as_static_str() {
+            return s.as_bytes();
+        }
+
```

Comment:
This is a performance sensitive method because it's how a user accesses the underlying string, so we want to make it as fast as possible. Adding an if-branch here will negatively effect performance.

Ideally when adding support for `&'static str` there shouldn't be any changes to this method, or to `fn len(...)` below, this should be achievable! Specifically the layout of `StaticStr` and `HeapBuffer` should both have the pointer as their first element and length as their second. Then we can kind of represent the `StaticStr` variant as a "heap variant with 0 capacity". This should allow us to keep branchless accesses.

Let me know what you think! We can merge this as-is and I can follow up with these changes, up to you 😊

Reply:
I'm quite busy recently and you are more familiar with compact-str, so I think merging it as-is and then improve it later might be faster and have less mental burden.

Reply:
I've optimized this by changing `StaticStr` layout to be more similiar to `HeapBuffer` so that we can employ the same optimization trick here.